src/main/java/org/skriptlang/skript/scheduler/Task.java

```diff
@@ -0,0 +1,140 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.scheduler;
+
+import org.bukkit.plugin.Plugin;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.Closeable;
+
+/**
+ * Represents a Task that only calls it's runnable synchronous.
+ * Unless otherwise defined by the scheduler like Folia.
+ * <p>
+ * See {@link AsyncTask} for asynchronous tasks.
+ */
+public abstract class Task implements Runnable, Closeable {
+
+	private final Plugin plugin;
+	private long periodInTicks = -1;
+
+	public Task(long delayInTicks, long periodInTicks) {
+		this(Skript.getInstance(), delayInTicks, periodInTicks);
+	}
+
+	public Task(Plugin plugin, long delayInTicks, long periodInTicks) {
+		this.plugin = plugin;
+		this.periodInTicks = periodInTicks;
+		TaskManager.run(this, delayInTicks);
+	}
+
+	public Task(long delayInTicks) {
+		this(Skript.getInstance(), delayInTicks);
+	}
+
+	public Task(Plugin plugin, long delayInTicks) {
+		this.plugin = plugin;
+		TaskManager.run(this, delayInTicks);
+	}
+
+	/**
+	 * @return The plugin that is calling this task.
+	 */
+	public Plugin getPlugin() {
+		return plugin;
+	}
+
+	/**
+	 * @return The period of this task if it's repeating in ticks, otherwise -1
+	 */
+	public long getPeriod() {
+		return periodInTicks;
+	}
+
+	/**
+	 * @return Whether this task is still running, i.e. whether it will run later or is currently running.
+	 */
+	public final boolean isAlive() {
+		return TaskManager.isAlive(this);
+	}
+
+	/**
+	 * Cancels this task.
+	 * @return true if the task was successfully cancelled.
+	 */
+	public final boolean cancel() {
+		return TaskManager.cancel(this);
+	}
+
+	@Override
+	public void close() {
+		cancel();
+	}
+
+	/**
+	 * Re-schedules the task to run next after the given delay. If this task was repeating it will continue so using the same period as before.
+	 * 
+	 * @param delay The delay using the existing TimeUnit.
```

Comment:
```suggestion
	 * Re-schedules the task to run next after the given delay. If this task was repeating it will continue so using the same period as before.
	 * 
	 * @param delay The delay in ticks.
```

Reply:
A task doesn't always mean it's related to ticks

Reply:
But all the logic is done in ticks? so therefore it is in ticks? This isn't AsyncTask and doesn't have all the timeunit stuff it does. Which begs the question as to why exactly it doesn't, but that's another thing.